src/snowflake/snowpark/functions.py

```diff
@@ -973,25 +974,16 @@ def create_map(
         <BLANKLINE>
     """
 
-    # Note: The type hint seems wrong here, hard to infer what the correct API here is.
+    check_create_map_parameter(*cols)
 
-    if len(cols) == 1 and isinstance(cols[0], (list, set)):
+    # TODO SNOW-1790918: Remove as part of refactoring with alias.
+    if len(cols) == 1 and isinstance(cols[0], (list, tuple)):
         cols = cols[0]
```

Comment:
Because we want to decouple the check from the internals :) In the end there will be on the thin client only "check + AST logic" left. If we move `cols=cols[0]` outside of the check, we don't have that clear separation.